lib/thrift/generator/utils.ex

```diff
@@ -149,4 +159,82 @@ defmodule Thrift.Generator.Utils do
     debug_optimization(expr, "done")
     expr
   end
+
+  @doc """
+  Returns a quoted value, resolving it in the schema if necessary
+
+  The values returned by this function are suitable for unquoting.
+  """
+  def quote_value(%ValueRef{} = ref, type, schema) do
```

Comment:
This function is just `StructGenerator.default_value/3` moved to Utils because I needed it to be public and usable by the constant generator.  I wasn't convinced that Utils was the best place for it to land but couldn't spot anything better - open to suggestions.

There is one new function clause that I commented on before (below in the file) that could be a bug wrt constant parsing, but I'm not sure.